### PR TITLE
Feature - Find Persons by Tag

### LINKED ISSUE #127: Feature - Find Tag command
This task will cover:
- [ ] Searching contacts using a given Tag
- [ ] Adding a command to find contacts by a given Tag

### PR DESCRIPTION
**Pull request by KwanHW**
# Changes
This PR adds the command to allow the user to find persons who contains a tag matching the specified keyword.

# Related Issues
- Closes #127

### CODE REVIEW

File: src/main/java/seedu/contax/logic/parser/FindByTagCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+public class FindByTagCommandParser implements Parser<FindByTagCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of FindByTagCommand and returns a FindByTagCommand
+     * object for execution.
+     */
+    public FindByTagCommand parse(String args) throws ParseException {
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize(args, PREFIX_TAG);
+
+        if (argumentMultimap.getValue(PREFIX_TAG).isEmpty()) {
```
Comment (sharpstorm) [inline src/main/java/seedu/contax/logic/parser/FindByTagCommandParser.java:22]: Is a preamble check necessary here?

File: src/main/java/seedu/contax/model/Model.java [hunk 2]
```diff
@@ -115,7 +118,9 @@
      */
     void setTag(Tag target, Tag editedTag);
 
-    ObservableList<Tag> getTagList();
+    void updateFilteredTagList(Predicate<Tag> predicate);
+
+    ObservableList<Tag> getFilteredTagList();
```
Comment (sharpstorm) [inline src/main/java/seedu/contax/model/Model.java:123]: I would appreciate if you could add javadocs.

**Review by sharpstorm — APPROVED** [review]
Looks good overall, less some minor issues.

File: src/main/java/seedu/contax/model/Model.java (hunk 2 above)
Comment (KwanHW) [reply src/main/java/seedu/contax/model/Model.java]: Fixed in 383c8d1a

File: src/main/java/seedu/contax/logic/parser/FindByTagCommandParser.java (hunk 1 above)
Comment (KwanHW) [reply src/main/java/seedu/contax/logic/parser/FindByTagCommandParser.java]: Fixed in 383c8d1a

**Review by dandaandaaaaaan — APPROVED** [review]
lgtm!

**Pull request merged**